client/src/stream.rs

```diff
@@ -248,16 +255,22 @@ impl<'ctx> ClientStream<'ctx> {
             user_ptr,
             token: data.token,
             device_change_cb,
+            shutdown_rx,
         }));
         Ok(unsafe { Stream::from_ptr(stream as *mut _) })
     }
 }
 
 impl<'ctx> Drop for ClientStream<'ctx> {
     fn drop(&mut self) {
-        debug!("ClientStream dropped...");
+        debug!("ClientStream drop");
         let rpc = self.context.rpc();
         let _ = send_recv!(rpc, StreamDestroy(self.token) => StreamDestroyed);
+        debug!("ClientStream drop - stream destroyed");
+        // Wait for CallbackServer to complete.  CallbackServer shutdown is triggered
+        // by the server dropped its side of the RPC connection.
+        let _ = self.shutdown_rx.recv();
```

Comment:
Not sure I understand this correctly. Should we call `_shutdown_tx.send(())` when dropping `CallbackServer`?

Will here get a `Err(RecvError)` and use it as the signal?

Reply:
> Will here get a Err(RecvError), caused by drop(_shutdown_tx) is run when dropping CallbackServer, and use it as the signal?

That's right. I didn't think it was worth adding a Drop implementation to CallbackServer to call send explicitly when we can rely on the error result from recv for all situations.  I'll clarify the comment (and fix the dropped -> dropping typo).

Reply:
Updated comment.

Reply:
Thanks for the explanation. Could you add some comments mentioning the code there takes an `Err(RecvError)` as a signal? I think it would be less confused.